### PR TITLE
refactor: improve docstrings and namings of the `Task` class

### PR DESCRIPTION
**Pull request by gcroci2**
I have not yet created another class to handle network-related parameters. Thinking about it, I am not convinced. It makes no sense to create a class whose only function is to create a few attributes in the init, and from what I understand this would be the case. Maybe a dictionary passed to the `Task` class would be more than enough for this case. I also opened an issue (#36) for evaluating Dynaconf - The configuration management package, which I still think is the best way to go. 
Thoughts? @DaniBodor

### CODE REVIEW

File: annubes/tasks/task.py [hunk 1]
```diff
@@ -16,24 +16,26 @@ class Task:
...
             Defaults to [0.8, 0.9, 1].
         catch_prob: probability of catch trials in the session, between 0 and 1.
             Defaults to 0.5.
-        catch_intensity: Intensity value during a catch trial.
-            Defaults to 0.
-        shuffle_trials: If True (default), trial order will be randomized. If False, all trials of one modality are run
-            before any trial of the next modality starts, in the order defined in `session` followed by catch trials.
+        shuffle_trials: If True (default), trial order will be randomized. If False, all trials corresponding to one
+            modality (e.g. visual) are run before any trial of the next modality (e.g. auditory) starts, in the order
+            defined in `session` followed by catch trials.
```
Comment (DaniBodor) [inline annubes/tasks/task.py:32]:
```suggestion
            defined in `session`, followed by catch trials.
```

File: annubes/tasks/task.py [hunk 2]
```diff
@@ -16,24 +16,26 @@ class Task:
 
     Args:
         name: Name of the task.
-        session: Dictionary representing the ratio (values) of the different trials (keys) within the task. Trials with
-            a single modality must be represented by single characters, while trials with multiple modalities are
-            represented by the character combination of those trials. Note that values are read relative to each other,
-            such that e.g. `{"v": 0.25, "a": 0.75}` == `{"v": 1, "a": 3}` is True. Defaults to {"v": 0.5, "a": 0.5}.
-        stim_intensities: List of possible intensities of each stimulus.
+        session: Structure of the type of trials which can appear during a session.
```
Comment (DaniBodor) [inline annubes/tasks/task.py:19]: I don't like using the word "type" here, as it has a very specific meaning in python, which is not the what we mean here. Also, it is not really a structure.

```suggestion
        session: Configuration of the trials that can appear during a session.
```

File: annubes/tasks/task.py [hunk 3]
```diff
@@ -16,24 +16,26 @@ class Task:
...
+            It is given by a dictionary representing the ratio (values) of the different trials (keys) within the task.
+            Trials with a single modality (e.g., a visual trial) must be represented by single characters, while trials
+            with multiple modalities (e.g., an audiovisual trial) are represented by the character combination of those
+            trials. Note that values are read relative to each other, such that e.g. `{"v": 0.25, "a": 0.75}` ==
+            `{"v": 1, "a": 3}` is True.
+            Defaults to {"v": 0.5, "a": 0.5}.
+        stim_intensities: List of possible intensity values of each stimulus type.
```
Comment (DaniBodor) [inline annubes/tasks/task.py:26]: Again, I would avoid the word "type".

```suggestion
        stim_intensities: List of possible intensity values of each stimulus.
```

File: annubes/tasks/task.py [hunk 4]
```diff
@@ -16,24 +16,26 @@ class Task:
...
         max_sequential: Maximum number of sequential trials of the same modality. Only used if shuffle is True.
             Defaults to None (no maximum).
-        n_outputs: Number of output signals that will be generated.
+        n_outputs: Number of output nodes in the network, signaling different behavioral choices.
             Defaults to 2.
-        output_intensities: List of possible intensities of the output signals. Currently only the smallest and largest
-            value of this list are used.
+        output_behavior: List of possible intensity values of the behavioral output. Currently only the smallest and
+            largest value of this list are used.
             Defaults to [0, 1].
```
Comment (DaniBodor) [inline annubes/tasks/task.py:39]: From what I understood, we decided to not use this after all, until we start setting up `Tasks` that actually will use this parameter. I think that should be done in a separate PR. Could you perhaps make an issue for that?

**Comment by DaniBodor** [comment]
> Thoughts? @DaniBodor

I'll reply on the issue in question, so that things are in the right place

**Comment by gcroci2** [comment]
> > > > Thoughts? @DaniBodor
> > > 
> > > 
> > > I'll reply on the issue in question, so that things are in the right place
> > 
> > 
> > I will also implement in this PR the separate class for the network-related parameters. I will then ask you to review it again :)
> 
> Why is that better than making new separate PR for that? I think it would be easier to keep track of what is going on if this one is merged as a "docs" PR, just with the changes to the docstring and nomenclature, and then make a separate PR for the implementation of the settings class(es).

Both PRs are relatively small, but okay I will do that.
